### PR TITLE
Mock/pool complete

### PR DESCRIPTION
**Pull request by Ben-PH**
Uses the automock crate to provide mock-stories for the pool tests. This removes the need for threading in many places, is more explicit about expected invocations, and the hard-coded results.
 

* [x] document all added functions
* [ ] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: massa-pool-worker/src/tests/scenario.rs [hunk 1]
```diff
@@ -32,113 +29,111 @@ use massa_models::test_exports::{
...
-/// that return the full storage)
-/// The execution thread will response that no operations had been executed.
-///
-/// ## Expected results
-/// The execution controller is expected to be asked 2 times for the first interaction:
-/// - to check the already executed operations
-/// - to check the final and candidate balances of the creator address
-/// And one time for the 9 next to check the executed operations.
+/// Create a mock-execution-controller story:
+/// 1. unexpected_opse_among, returning storage.get_op_refs()
```
Comment (Ben-PH) [inline massa-pool-worker/src/tests/scenario.rs:46]: this needs to be updated

File: massa-pool-worker/src/tests/scenario.rs [hunk 2]
```diff
@@ -338,94 +320,79 @@ fn test_denunciation_pool_get() {
...
+) -> Box<MockSelectorController> {
+    let mut selector_controller = Box::new(MockSelectorController::new());
+    selector_controller
+        .expect_clone_box()
+        .times(2)
+        .returning(move || Box::new(MockSelectorController::new()));
+    selector_controller
+        .expect_clone_box()
+        .return_once(move || Box::new(story));
+    selector_controller
```
Comment (Ben-PH) [inline massa-pool-worker/src/tests/scenario.rs:397]: double-check it's being used wherever it makes sense

File: massa-pool-worker/src/tests/tools.rs [hunk 3]
```diff
@@ -84,40 +81,36 @@ pub(crate) fn create_some_operations(n: usize, op_gen: &OpGenerator) -> Vec<Secu
...
-        Receiver<MockExecutionControllerMessage>,
-        crossbeam_channel::Receiver<MockSelectorControllerMessage>,
-        Storage,
-    ),
-{
+pub fn pool_test(
+    cfg: PoolConfig,
+    execution_story: Box<MockExecutionController>,
+    selector_story: Box<AutoMockSelectorController>,
+) -> PoolTestBoilerPlate {
```
Comment (Ben-PH) [inline massa-pool-worker/src/tests/tools.rs:94]: doc-comment

**Review by AurelienFT — APPROVED** [review]

**Review by AurelienFT — APPROVED** [review]
Really like this crate. Nice PR. Thank you !

**Pull request merged**